mozci/push.py

```diff
@@ -1146,11 +1151,16 @@ def __make_group_summary_objects(pushes):
 
     for task_id, result_group, result_duration in results:
         # Obtain TestTask object that matches the task_id.
-        task = [t for push in pushes for t in push.tasks if t.id == task_id].pop()
+        tasks = [t for push in pushes for t in push.tasks if t.id == task_id]
+        if len(tasks) == 0:
+            continue
+        task = tasks.pop()
 
+        parts = task.label.split('/')
+        key = "%s-%s:%s" % (parts[0], parts[1].split('-')[0], result_group)
         # Build the mapping of group to the group durations and TestTask objects.
-        mapping[result_group]["tasks"].append(task)
-        mapping[result_group]["durations"].append(result_duration)
+        mapping[key]["tasks"].append(task)
+        mapping[key]["durations"].append(result_duration)
 
     return [
         GroupSummary(key, value["tasks"], value["durations"])
```

Comment:
`key` here is no longer a group now, I'm not sure this will work

Reply:
Ah, I see you changed the meaning of a GroupSummary.
Why though? We could keep the previous meaning, and have a GroupSummary for each group on a set of tasks.
E.g., to simplify, a GroupSummary for a given group for all Windows task, a GroupSummary for a given group for all Linux tasks, and so on.